libs/wallet/src/order-hooks/use-order-cancel.tsx

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useEffect, useState } from 'react';
+import { determineId } from '@vegaprotocol/react-helpers';
+
+import { useVegaTransaction } from '../use-vega-transaction';
+import { useVegaWallet } from '../use-vega-wallet';
+import { useSubscription } from '@apollo/client';
+import type {
+  OrderEvent,
+  OrderEventVariables,
+  OrderEvent_busEvents_event_Order,
+} from './__generated__/OrderEvent';
+import { ORDER_EVENT_SUB } from './order-event-query';
+import * as Sentry from '@sentry/react';
+
+export const useOrderCancel = () => {
+  const { keypair } = useVegaWallet();
+  const { send, transaction, reset: resetTransaction } = useVegaTransaction();
+  const [finalizedOrder, setFinalizedOrder] =
+    useState<OrderEvent_busEvents_event_Order | null>(null);
+  const [id, setId] = useState('');
+
+  // Start a subscription looking for the newly created order
+  useSubscription<OrderEvent, OrderEventVariables>(ORDER_EVENT_SUB, {
+    variables: { partyId: keypair?.pub || '' },
+    skip: !id,
+    onSubscriptionData: ({ subscriptionData }) => {
+      if (!subscriptionData.data?.busEvents?.length) {
+        return;
+      }
+      // No types available for the subscription result
+      const matchingOrderEvent = subscriptionData.data.busEvents[0].event;
+
+      if (matchingOrderEvent && matchingOrderEvent.__typename === 'Order') {
+        setFinalizedOrder(matchingOrderEvent);
+      }
+    },
+  });
+
+  useEffect(() => {
+    if (finalizedOrder) {
+      resetTransaction();
```

Comment:
question (blocking): @mattrussell36 I notice this is also present in the useSubmitOrder function so asking you haha, but this isn't cleaning up the subscription after the event has been fired. I know this is in a hook so once the element in unrendered the subscription will be cancelled but IMO we should lean up the listener here as well. Was there a reason we decided not to do this in useOrderSubmit?

It's not the worst thing in the world as this is filtered by partyId but wanted to double check. I just worry if this is used as part of non-transient UI we could end up with a memory leek, albeit a slow one

Reply:
chore (blocking): Additionally, not sure why I used a hook to reset transaction here it just makes things more confusing. Can we move resetTransaction() to just be called immediately after setUpdatedOrder. The code is more readable that way. Also could you make the same change in the use-order-submit hook.

Reply:
Updated to use resetTransaction immediately after setUpdatedOrder. 
Let me use sth else instead of useSubscription in another PR. I will refactor a bit order-list and move all the order hooks from the wallet.

Reply:
https://github.com/vegaprotocol/frontend-monorepo/issues/657